src/traces/table/plot.js

```diff
@@ -0,0 +1,892 @@
+
+    var columnCells = renderColumnCells(columnBlock);
+
+    var columnCell = renderColumnCell(columnCells);
+
+    setFont(columnCell);
+
+    var cellRect = renderCellRect(columnCell);
+
+    sizeAndStyleRect(cellRect);
+
+    var cellTextHolder = renderCellTextHolder(columnCell);
+
+    var cellText = renderCellText(cellTextHolder);
+
+    populateCellText(cellText, tableControlView, allColumnBlock, gd);
+
+    // doing this at the end when text, and text stlying are set
+    setCellHeightAndPositionY(columnCell);
+}
+
+function renderColumnCells(columnBlock) {
+
+    var columnCells = columnBlock.selectAll('.columnCells')
+        .data(gup.repeat, gup.keyFun);
+
+    columnCells.enter()
+        .append('g')
+        .classed('columnCells', true);
+
+    columnCells.exit()
+        .remove();
+
+    return columnCells;
+}
+
+function renderColumnCell(columnCells) {
+
+    var columnCell = columnCells.selectAll('.columnCell')
+        .data(splitToCells, function(d) {return d.keyWithinBlock;});
+
+    columnCell.enter()
+        .append('g')
+        .classed('columnCell', true);
+
+    columnCell.exit()
+        .remove();
+
+    return columnCell;
+}
+
+function renderCellRect(columnCell) {
+
+    var cellRect = columnCell.selectAll('.cellRect')
+        .data(gup.repeat, function(d) {return d.keyWithinBlock;});
+
+    cellRect.enter()
+        .append('rect')
+        .classed('cellRect', true);
+
+    return cellRect;
+}
+
+function renderCellText(cellTextHolder) {
+
+    var cellText = cellTextHolder.selectAll('.cellText')
+        .data(gup.repeat, function(d) {return d.keyWithinBlock;});
+
+    cellText.enter()
+        .append('text')
+        .classed('cellText', true);
+
+    return cellText;
+}
+
+function renderCellTextHolder(columnCell) {
+
+    var cellTextHolder = columnCell.selectAll('.cellTextHolder')
+        .data(gup.repeat, function(d) {return d.keyWithinBlock;});
+
+    cellTextHolder.enter()
+        .append('g')
+        .classed('cellTextHolder', true);
+
+    return cellTextHolder;
+}
+
+function setFont(columnCell) {
+    columnCell
+        .each(function(d, i) {
+            var spec = d.calcdata.cells.font;
+            var col = d.column.specIndex;
+            var font = {
+                size: gridPick(spec.size, col, i),
+                color: gridPick(spec.color, col, i),
+                family: gridPick(spec.family, col, i)
+            };
+            Drawing.font(d3.select(this), font);
+
+            d.rowNumber = d.key;
+            d.align = gridPick(d.calcdata.cells.align, col, i);
+            d.valign = gridPick(d.calcdata.cells.valign, col, i);
+            d.cellBorderWidth = gridPick(d.calcdata.cells.line.width, col, i);
+            d.font = font;
+        });
+}
+
+function sizeAndStyleRect(cellRect) {
+    cellRect
+        .attr('width', function(d) {return d.column.columnWidth;})
+        .attr('stroke-width', function(d) {return d.cellBorderWidth;})
```

Comment:
Please use [`Color.stroke`](https://github.com/plotly/plotly.js/blob/8b35a8e0a0484466d1f49cde769321aa6a120a35/src/components/color/index.js#L84) :recycle: whenever you can. That split between `stroke` and `stroke-opacity` is for Adobe Illustrator compat I believe.